### PR TITLE
Enable MySQL as an database alternative

### PR DESCRIPTION
**Pull request by invisiblek**
This pull request includes a

- [ ] Bug fix
- [x] New feature
- [ ] Translation

The following changes were made

-
-
-

If this is related to an existing ticket, include a link to it as well.

### CODE REVIEW

**Comment by shivampaw** [comment]
So renaming mysql.example.json to mysql.json and inputting values will automatically use mysql instead of sqlite?

Assume this works with remote mysql servers as well?

**Comment by Ashex1** [comment]
Since you're cleaning up #1223 can you make a PR against that fork?

https://github.com/vaurdan/PokemonGo-Map

**Comment by Bakies** [comment]
Related: #744 #1028

File: .gitignore [hunk 1]
```diff
@@ -3,7 +3,7 @@
 *.db
 *.db-journal
 *.sass-cache
-credentials.json
+*.json
```
Comment (chuyskywalker) [inline .gitignore]: Won't this exclude `package.json` which is needed for node stuff?

File: mysql.json.example [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+	"mysql_user"		: "",
+	"mysql_db"		: "",
+	"mysql_pass"		: "",
+	"mysql_host"		: ""
```
Comment (chuyskywalker) [inline mysql.json.example]: nit: I would think the `mysql_` prefix here is redundant.

File: pogom/utils.py [hunk 3]
```diff
@@ -104,15 +104,10 @@ def get_pokemon_name(pokemon_id):
 
     return get_pokemon_name.names[str(pokemon_id)]
 
-def load_credentials(filepath):
+def load_config(filepath):
```
Comment (chuyskywalker) [inline pogom/utils.py]: Both usages, and in theory all future usages of, `load_config` have to do the same boiler plate:

```
load_config(os.path.dirname(os.path.realpath(__file__))+os.path.sep+'file-name-here')
```

Why not code it to assume a prefix of `os.path.dirname(os.path.realpath(__file__))+os.path.sep` so people can just:

```
load_config('file-name-here')
```

Would be a lot less repeated code and easier to read through.

**Comment by Bitti09** [comment]
pls rebase :heart:

**Comment by vinsilsim** [comment]
@invisiblek 
Dont work

2016-07-23 01:40:10,528 [     models] [   INFO] Connecting to MySQL database on
localhost.
2016-07-23 01:40:11,980 [  runserver] [   INFO] Parsed location is: 34.0092/-118
.4976/0.0000 (lat/lng/alt)
2016-07-23 01:40:11,980 [     search] [   INFO] Map iteration: 0
2016-07-23 01:40:11,980 [     search] [   INFO] Attempting login to Pokemon Go.
2016-07-23 01:40:11,980 [   auth_ptc] [   INFO] PTC login for: vinsilsim
2016-07-23 01:40:12,013 [  _internal] [   INFO]  * Running on http://172.31.1.10
0:2222/ (Press CTRL+C to quit)
2016-07-23 01:40:12,684 [   auth_ptc] [   INFO] PTC Login successful
2016-07-23 01:40:13,010 [     search] [   INFO] Login to Pokemon Go successful.
2016-07-23 01:40:13,010 [     search] [   INFO] Scanning step 1 of 100.
Exception in thread search_thread:
Traceback (most recent call last):
  File "C:\Python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "C:\Python27\lib\threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "C:\Users\Administrator\Desktop\PokemonMap_Mysql\pogom\search.py", line 1
06, in search_loop
    search(args, i)
  File "C:\Users\Administrator\Desktop\PokemonMap_Mysql\pogom\search.py", line 8
7, in search
    parse_map(response_dict, i, step)
TypeError: parse_map() takes exactly 1 argument (3 given)

**Comment by chuyskywalker** [comment]
@invisiblek -- if you're humped for time, I could pick this one up. Let me know.

**Pull request closed without merging**